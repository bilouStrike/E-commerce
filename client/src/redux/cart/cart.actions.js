import cartActionsTypes from './cart.types';

export const toogleCartHidden = () => ({
    type: cartActionsTypes.TOGGLE_CART_HIDDEN 
})

export const addItemToCart = item => ({
    type: cartActionsTypes.ADD_ITEM_TO_CART,
    payload: item
})

export const removeCartItem = item => ({
    type: cartActionsTypes.REMOVE_CART_ITEM,
    payload: item
})

export const clearCart = () => ({
    type: cartActionsTypes.CLEAR_CART
})