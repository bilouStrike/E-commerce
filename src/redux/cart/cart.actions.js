import cartActionsTypes from './cart.types';

export const toogleCartHidden = () => ({
    type: cartActionsTypes.TOGGLE_CART_HIDDEN 
})
