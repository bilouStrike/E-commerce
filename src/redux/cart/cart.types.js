const cartActionsTypes = {
    TOGGLE_CART_HIDDEN: 'TOOGLE_CART_HIDDEN',
    ADD_ITEM_TO_CART: 'ADD_ITEM_TO_CART',
    REMOVE_CART_ITEM: 'REMOVE_CART_ITEM'
}
export default cartActionsTypes;