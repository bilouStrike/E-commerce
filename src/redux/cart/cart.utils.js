const addItemToCart = ( cartItems, cartItemToAdd ) => {

    console.log(cartItems);
    const isExist = cartItems.find( item => item.id == cartItemToAdd.id );

    if( isExist ) {
       return cartItems.map( item =>  
            item.id === cartItemToAdd.id ? 
            { ...item, qunatity: item.qunatity + 1  } : item )
    } 

    return [ ...cartItems, {...cartItemToAdd, qunatity:1 } ]
}
export default addItemToCart;