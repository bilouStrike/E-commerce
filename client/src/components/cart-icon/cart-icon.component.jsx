import React from 'react';
import { connect } from 'react-redux';
import { toogleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';
import { ReactComponent as ShopingIcon } from '../../assets/shopping-bag.svg.svg';

import './cart-icon.styles.scss';

export const CartIcon = ({ toogleCartHidden, itemCount }) => (
  <div id="cart-icon" className="cart-icon" onClick={toogleCartHidden}>
    <ShopingIcon className="shopping-icon" />
    <span className="item-count"> {itemCount} </span>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  toogleCartHidden: () => dispatch(toogleCartHidden()),
});

const mapStateToProps = (state) => ({
  itemCount: selectCartItemsCount(state),
});
export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
