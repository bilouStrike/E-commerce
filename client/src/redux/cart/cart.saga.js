import { call, all, put, takeLatest } from 'redux-saga/effects';
import userActionsTypes from './../user/user.types';
import { clearCart } from './cart.actions';

export function* cleanCartOnSignOut() {
    yield put(clearCart());
}

export function* onSignOutSuccess() {
    yield takeLatest(userActionsTypes.SIGN_OUT_SUCCESS, cleanCartOnSignOut)
}

export function* cartSaga() {
    yield all([
        call(onSignOutSuccess)
    ]);
}