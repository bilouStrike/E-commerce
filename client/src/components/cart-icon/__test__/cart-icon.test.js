import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { CartIcon } from '../cart-icon.component';
import configureStore from 'redux-mock-store';
import { toogleCartHidden } from '../../../redux/cart/cart.actions';

const mockStore = configureStore();
it('expect to cart icon component', () => {
    const mocktoogleCartHidden = jest.fn();
    const itemCount = 3;
    const wrapper = toJson(shallow(<CartIcon toogleCartHidden={mocktoogleCartHidden}  itemCount={itemCount}/>));
    expect(wrapper).toMatchSnapshot();
});

/*it('expect to run toogleCart correctly', () => {
    const mocktoogleCartHidden = jest.fn();
    const mockitemCount = 3;
    const initialState = {
        hidden: true
    }
    const store = mockStore(initialState);
    store.dispatch = jest.fn();
    //store.dispatch(toogleCartHidden());
    const action = toogleCartHidden();
    const wrapper = mount(
            <CartIcon 
                store={store}
                toogleCartHidden={toogleCartHidden}
                itemCount={mockitemCount} />
            );
        console.log(wrapper.find('[id="cart-icon"]'));
    wrapper.find('[id="cart-icon"]').simulate('click');
    expect(store.dispatch).toHaveBeenCalled();
});*/