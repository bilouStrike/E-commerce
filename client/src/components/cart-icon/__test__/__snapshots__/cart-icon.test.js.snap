// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to cart icon component 1`] = `
<div
  className="cart-icon"
  id="cart-icon"
  onClick={[MockFunction]}
>
  <ForwardRef(SvgShoppingBagSvg)
    className="shopping-icon"
  />
  <span
    className="item-count"
  >
     
    3
     
  </span>
</div>
`;
